modules/lazyload/module.php

```diff
@@ -38,38 +38,146 @@ private function enqueue_styles() {
+			return false;
+		}
+
+		$repeater_selector = Utils::get_array_value_by_keys( $lazyload_options, [ 'selector' ] );
+		if ( ! $repeater_selector ) {
+			return false;
+		}
+
+		$repeater_keys = Utils::get_array_value_by_keys( $repeater_settings, [ 'background_lazyload', 'keys' ] );
+		$repeater_background_image_url = Utils::get_array_value_by_keys( $repeater_settings, $repeater_keys );
+
+		if ( $repeater_background_image_url ) {
+			return $repeater_selector;
+		}
+
+		return false;
+	}
+
+	private function get_controls_with_background_image( $controls ) {
+		return array_filter( $controls, function( $control ) {
+			return Utils::get_array_value_by_keys( $control, [ 'background_lazyload', 'active' ] );
+		} );
+	}
+
+	private function has_background_image( $control_data, $settings ) {
+
+		$keys = Utils::get_array_value_by_keys( $control_data, [ 'background_lazyload', 'keys' ] );
+		$background_image_controls = array_filter( array_keys( $settings ), function( $key ) use ( $keys ) {
+			return false !== strpos( $key, $keys[0] );
+		} );
+
+		foreach ( $background_image_controls as $background_image_control ) {
+			$breakpoint_background_keys = $keys;
+			$breakpoint_background_keys[0] = $background_image_control;
+			$background_image_url = Utils::get_array_value_by_keys( $settings, $breakpoint_background_keys );
+			if ( $background_image_url ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
 	private function update_element_attributes( Element_Base $element ) {
+
 		$settings = $element->get_settings_for_display();
 		$controls = $element->get_controls();
-		$lazyload_attribute_name = 'data-e-bg-lazyload';
+		$controls_with_background_image = $this->get_controls_with_background_image( $controls );
+		$repeaters_selectors = $this->get_repeaters_selectors( $controls, $settings );
 
-		$controls_with_background_image = array_filter( $controls, function( $control ) {
-			return Utils::get_array_value_by_keys( $control, [ 'background_lazyload', 'active' ] );
-		} );
+		$lazyload_attribute_name = 'data-e-bg-lazyload';
+		$attributes = [];
 
 		foreach ( $controls_with_background_image as $control_name => $control_data ) {
-			$keys = Utils::get_array_value_by_keys( $control_data, [ 'background_lazyload', 'keys' ] );
-			$background_image_url = Utils::get_array_value_by_keys( $settings, $keys );
-			if ( $background_image_url ) {
 
-				$has_attribute = $element->get_render_attributes( '_wrapper', $lazyload_attribute_name );
-				if ( ! $has_attribute ) {
-					$bg_selector = Utils::get_array_value_by_keys( $control_data, [ 'background_lazyload', 'selector' ] ) ?? '';
-					$element->add_render_attribute( '_wrapper', [
-						$lazyload_attribute_name => $bg_selector,
-					] );
+			$control_selectors = [];
+
+			// Marge repeaters selectors to the wrapper control selectors if exist.
+			if ( $repeaters_selectors ) {
+				$control_selectors = array_merge( $repeaters_selectors, $control_selectors );
+			}
```

Comment:
Can it move outside of the loop

Reply:
Moving the $selectors = []; statement outside the loop would cause the $selectors array to retain the values from the previous iteration. This would result in incorrect values being added to the $control_selectors array for each iteration.

In the current implementation, the $selectors array is initialized as an empty array at the start of each iteration of the loop. This ensures that the $control_selectors array only contains selectors from the current iteration and not from previous iterations.

If you move the $selectors = []; statement outside the loop, then the $control_selectors array would accumulate selectors from all iterations of the loop. This would result in incorrect values being added to the $attributes array for the lazyload attribute.

Reply:
Update - We sit together and refactor the code.